sky/cli.py

```diff
@@ -1457,6 +1476,11 @@ def exec(
         env=env,
     )
 
+    if isinstance(task_or_dag, sky.Dag):
+        raise click.UsageError('YAML specifies a DAG, while `sky exec` '
```

Comment:
'YAML specifies a DAG which is only supported by `sky spot launch`. `sky exec` '

Reply:
Done. Thanks!